src/lib.rs

```diff
@@ -467,8 +469,8 @@ impl GenerateTasksService for GenerateTasksServiceImpl {
             for task in &build_variant.tasks {
                 // Burn in tasks could be different for each build variant, so we will always
                 // handle them.
-                if task.name == BURN_IN_TESTS {
-                    if self.gen_burn_in {
+                if self.gen_burn_in {
```

Comment:
Can you add some tests for this logic to make sure we're skipping the tasks when we should be?

Reply:
This particular function seems hard to test or split. I've added end-to-end test for this logic.

Also added a comment to make it more clear:
https://github.com/mongodb/mongo-task-generator/pull/47/files#diff-7ff815b68943f08bef79f083c09e55ca8c18db8e25033034b12fa31f1d43dab4R64-R65

Let me know if you have better ideas.